### PR TITLE
Eslint config validation

### LINKED ISSUE #85: Trouble with .graphqlconfig / bad config validation / questions
Sorry for bundling several things into one issue. I initally had just one problem, but when diagnosing it things snowballed.

TL;DR: This plugin works super well, just need to look over config validation in general .)

Here's the test repo: https://github.com/lydell/eslint-plugin-graphql-bug

### `graphql/named-operations` .graphqlconfig problem

I tried switching from `schemaJson: require("./schema.json")` to using `.graphqlconfig`. This worked well for the `graphql/template-strings` rule, but I got trouble with the `graphql/named-operations` rule.

Using `.graphqlconfig` it looks as if no additional config is need for the `graphql/named-operations` rule in `.eslintrc.js`. So I tried this:

```
"graphql/named-operations": "error",
```

But then nothing is reported. This works, though:

```
"graphql/named-operations": ["error", {}],
```

### `graphql/capitalized-type-name` .graphqlconfig problem

Same thing here. It looks as if this should be enough:

```
"graphql/capitalized-type-name": "error",
```

This results in a validation error:

```
"graphql/capitalized-type-name": ["error", {}],
```
```
Configuration for rule "graphql/capitalized-type-name" is invalid:
  Value "[object Object]" should have required property '.schemaJson'.
  Value "[object Object]" should have required property '.schemaJsonFilepath'.
  Value "[object Object]" should have required property '.schemaString'.
  Value "[object Object]" should match exactly one schema in oneOf.
```

I couldn't get this rule to work .graphqlconfig.

### Questions

Diagnosing the above problems and looking at the other rules, I stumbled upon some questions:

- Why does `graphql/named-operations` require a schema?
- Why does `graphql/capitalized-type-name` require a schema?
- Why does `graphql/required-fields` let you specify env?
  - It seems to be optional, so what's the default?
- Is env required for `graphql/template-strings`? It does not seem so when I
  tested. What's the default?

### Other minor config validation quirks

They're all described in https://github.com/lydell/eslint-plugin-graphql-bug

### PR DESCRIPTION
**Pull request by lydell**
TODO:

- [x] Make sure all of the significant new logic is covered by tests
- [x] Rebase your changes on master so that they can be merged easily
- [x] Make sure all tests pass
- [x] Update CHANGELOG.md with your change
- [ ] If this was a change that affects the external API, update the README

Fixes #85

### CODE REVIEW

**Comment by lydell** [comment]
@stubailo @jnwng This is supposed to be an easy-to-merge non-controversial PR that fixes a couple of smaller bugs. Most importantly, it makes .graphqlconfig work with the graphql/capitalized-type-name rule. Is there anything you'd like me to change before merging? :)

**Comment by jnwng** [comment]
I'll take a look when I get on the train, thanks for reminding me!

File: README.md [hunk 1]
```diff
@@ -246,6 +270,24 @@ module.exports = {
 }
 ```
 
+When using `.graphqlconfig`, you might end up not needing to specify any additional options for a rule. Note that you still _have_ to provide an empty object (`{}`) to the rule:
```
Comment (jnwng) [inline README.md]: this API is a little strange to suggest. [can we default the optionGroup to an empty object?](https://github.com/apollographql/eslint-plugin-graphql/blob/master/src/index.js#L246-L254) that should (hopefully) resolve some of this wonkiness

Comment (lydell) [reply README.md]: I'll see if I can come up with something :)

Comment (lydell) [reply README.md]: I pushed a fix now. (Note that this would also be fixed by #87).

**Comment by lydell** [comment]
@jnwng Is there anything else you’d like me to do in this PR? :)

**Comment by jnwng** [comment]
nope, just maybe find me more time in the day :)

thank you @lydell!

**Pull request merged**